### DISCUSSION TITLE
Incorrect Screen Resolution on 4K Android TV Emulator ?

### DISCUSSION THREAD

**Discussion by hersentino** [react-native-tvos/react-native-tvos#604] [Q&A] [answered]
Hi, 

When I create a new React Native project using react-native-tv-os and display the screen resolution (using [Dimensions](https://reactnative.dev/docs/dimensions)), the result on a 4K android TV emulator is `width: 960px, height: 480px`. I would expect it to be 3840x2160. Does anyone know why or have any suggestions on how to fix it?

Thanks !

**Comment by douglowder** [comment, marked as answer]
I took the standard new project example app, and added code to display dimensions:

```
  const {height, width, scale, fontScale} = useWindowDimensions();
```

Here are the results for various simulators and emulators. It appears that Android TV always gives the same result regardless of screen size. The Apple TV results seem correct.

The actual native code that gets the dimensions is in `ReactAndroid/src/main/java/com/facebook/react/uimanager/DisplayMetricsHolder.java`, and is calling `context.getResources().getDisplayMetrics()`.

I'll have to dig into this to see if this is really what the OS is returning (we can't fix that), or if somehow a default value is always being returned (which we can fix).

| Device  	| Height   	| Width  	| Scale  	| Font Scale  	|
|---	|---	|---	|---	|---	|
|  Apple TV 	|  1080 	| 1920  	| 1  	| 1  	|
|  Apple TV 4K 	|  1080 	| 1920  	|  2 	|  1 	|
|  Android TV 1080p 	| 540   	| 960  	| 2  	| 1  	|
|  Android TV 4K 	|  540 	|  960 	| 2  	|   1	|
|  Sony Bravia OLED TV 	| 540  	| 960  	| 2  	| 1  	|

**Reply by hersentino** [reply]
Thank you for looking into the issue. Indeed, for the Apple TV, I confirm that everything is working fine, whether it's in 1920 or in 4K, the image is clear. The problem seems to be related only to Android TV.
